src/backend/aspen/api/utils/pathogens.py

```diff
@@ -0,0 +1,50 @@
+import sqlalchemy as sa
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import joinedload
+from sqlalchemy.sql.expression import and_
+
+from aspen.database.models import Pathogen, PathogenRepoConfig, PublicRepository
+
+
+async def get_public_repository_prefix(pathogen: Pathogen, public_repository_name, db):
+    prefix = (
+        sa.select(PathogenRepoConfig)  # type: ignore
+        .join(PublicRepository)  # type: ignore
+        .where(  # type: ignore
+            and_(
+                PathogenRepoConfig.pathogen == pathogen,
+                PublicRepository.name == public_repository_name,
+            )
+        )
+    )
+    res = await db.execute(prefix)
+    pathogen_repo_config = res.scalars().one_or_none()
+
+    if pathogen_repo_config:
+        return pathogen_repo_config.prefix
+
+
+async def get_pathogen_repo_config_for_pathogen(
+    pathogen: Pathogen, preferred_public_database: str, db: AsyncSession
+):
+
+    q = (
+        sa.select(PathogenRepoConfig)  # type: ignore
+        .options(  # type: ignore
+            joinedload(PathogenRepoConfig.public_repository),
+            joinedload(PathogenRepoConfig.pathogen),
+        )
+        .where(
+            and_(
+                PathogenRepoConfig.pathogen == pathogen,
+                PathogenRepoConfig.public_repository.has(
+                    PublicRepository.name == preferred_public_database
+                ),
+            )
+        )
+    )
+    res = await db.execute(q)
+    pathogen_repo_config = res.scalars().one_or_none()
+
+    if pathogen_repo_config:
+        return pathogen_repo_config
```

Comment:
I think this `if` doesn't do anything -- pathogen_repo_config is already an object or None

Reply:
true, i'll take it out